hubblestack/files/hubblestack_nova/misc.py

```diff
@@ -767,22 +708,241 @@ def ensure_reverse_path_filtering(reason=''):
+            path=path,
+        )
+    )
+
+    try:
+        ret = __salt__['cmd.run_all'](cmd, python_shell=False, ignore_retcode=True)
+    except (IOError, OSError) as exc:
+        raise CommandExecutionError(exc.strerror)
+
+    return ret
+
+
+def check_list_values(file_path, match_pattern, value_pattern, grep_arg, white_list, black_list, value_delimter):
+    '''
+    This function will first get the line matching given match_pattern.
+    After this value pattern will be extracted from the above line.
+    value pattern will be splitted by value_delimiter to get the list of values.
+    match_pattern will be regex patter for grep command.
+    value_pattern will be regex for re module of python to get matched values.
+    Only one of white_list and blacklist is allowed.
+    white_list and black_list should have comma(,) seperated values.
+
+    Example for CIS-2.2.1.2
+    ensure_ntp_configured:
+      data:
+        CentOS Linux-7:
+         tag: 2.2.1.2
+         function: check_list_values
+         args:
+           - /etc/ntp.conf
+           - '^restrict.*default'
+           - '^restrict.*default(.*)$'
+           - null
+           - kod,nomodify,notrap,nopeer,noquery
+           - null
+           - ' '
+         description: Ensure ntp is configured
+    '''
+
+    list_delimter = ","
+
+    if black_list is not None and white_list is not None:
+        return "Both black_list and white_list values are not allowed."
+
+    grep_args = [] if grep_arg is None else [grep_arg]
+    matched_lines = _grep(file_path, match_pattern, *grep_args).get('stdout')
+    if not matched_lines:
+        return "No match found for the given pattern: " + str(match_pattern)
+
+    matched_lines = matched_lines.split('\n') if matched_lines is not None else []
+    error = []
+    for matched_line in matched_lines:
+        regexp = re.compile(value_pattern)
+        matched_values = regexp.search(matched_line).group(1)
+        matched_values = matched_values.strip().split(value_delimter) if matched_values is not None else []
+        if white_list is not None:
+            values_not_in_white_list = list(set(matched_values) - set(white_list.strip().split(list_delimter)))
+            if values_not_in_white_list != []:
+                error += ["values not in whitelist: " + str(values_not_in_white_list)]
+        else:
+            values_in_black_list = list(set(matched_values).intersection(set(black_list.strip().split(list_delimter))))
+            if values_in_black_list != []:
+                error += ["values in blacklist: " + str(values_in_black_list)]
+
+    return True if error == [] else str(error)
+
+
+def mail_conf_check(reason=''):
+    '''
+    Ensure mail transfer agent is configured for local-only mode
+    '''
+    valid_addresses = ["localhost", "127.0.0.1", "::1"]
+    mail_addresses = _execute_shell_command("grep '^[[:blank:]]*inet_interfaces' /etc/postfix/main.cf | awk -F'=' '{print $2}'").strip()
```

Comment:
Specific for systems that have postfix installed, but not generic enough for all MTAs. This probably should be indicated somehow.

Reply:
Opened a new ticket to handle this case.